internal/tiltfile/tiltfile_state.go

```diff
@@ -1331,28 +1331,44 @@ type k8sObjectSelector struct {
 // Creates a new k8sObjectSelector
 // If an arg is an empty string it will become an empty regex that matches all input
 // Otherwise the arg must match the input exactly
-func newExactK8sObjectSelector(apiVersion string, kind string, name string, namespace string) (k8sObjectSelector, error) {
-	ret, err := newK8sObjectSelector(
-		exactOrEmptyRegex(apiVersion),
-		exactOrEmptyRegex(kind),
-		exactOrEmptyRegex(name),
-		exactOrEmptyRegex(namespace),
-	)
+func newExactCaseInsensitiveK8sObjectSelector(apiVersion string, kind string, name string, namespace string) (k8sObjectSelector, error) {
+	ret := k8sObjectSelector{apiVersionString: apiVersion, kindString: kind, nameString: name, namespaceString: namespace}
+	var err error
+
+	ret.apiVersion, err = regexp.Compile(exactOrEmptyRegex(apiVersion))
+	if err != nil {
+		return k8sObjectSelector{}, errors.Wrap(err, "error parsing apiVersion regexp")
+	}
+
+	ret.kind, err = regexp.Compile(exactOrEmptyRegex(kind))
+	if err != nil {
+		return k8sObjectSelector{}, errors.Wrap(err, "error parsing kind regexp")
+	}
+
+	ret.name, err = regexp.Compile(exactOrEmptyRegex(name))
 	if err != nil {
-		return ret, err
+		return k8sObjectSelector{}, errors.Wrap(err, "error parsing name regexp")
 	}
 
-	ret.apiVersionString = apiVersion
-	ret.kindString = kind
-	ret.nameString = name
-	ret.namespaceString = namespace
+	ret.namespace, err = regexp.Compile(exactOrEmptyRegex(namespace))
+	if err != nil {
+		return k8sObjectSelector{}, errors.Wrap(err, "error parsing namespace regexp")
+	}
 
 	return ret, nil
 }
 
+func makeCaseInsensitive(s string) string {
```

Comment:
i feel like half the problem here is that we're passing around a regexp around as a DSL string, rather than something more structured, and i wonder if this code would be easier to follow if we had something like

```
type re struct {
  pattern string
  ignoreCase bool
  fullMatch bool
}

func (r re) compile() { ... }
```

Reply:
Yeahhh I tried something like this back when I was originally writing this a couple weeks ago. I'll give it another shot with fresh eyes.